### ISSUE TITLE
crash after installing sli.dev

### ISSUE THREAD

**Issue by draptik** [athityakumar/colorls#461]
Installing sli.dev, and then switching to the presenation folder, crashes colorls.

- Type of issue :
  - [x] Bug in existing feature

## Error message

```sh
❯ lc
/usr/lib/ruby/3.0.0/set.rb:355:in `add': can't modify frozen Hash: {"??"=>true} (FrozenError)
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/git.rb:22:in `block in status'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/git.rb:66:in `block in git_subdir_status'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/git.rb:59:in `popen'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/git.rb:59:in `git_subdir_status'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/git.rb:17:in `status'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:124:in `block in init_git_status'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:247:in `git_info'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:316:in `fetch_string'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:323:in `block in ls_line'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `each'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `each_with_index'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `ls_line'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:92:in `block in ls'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `block in each_line'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each_slice'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each_line'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:91:in `ls'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/core.rb:59:in `ls_dir'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/flags.rb:86:in `block in process_args'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/flags.rb:83:in `each'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/flags.rb:83:in `process_args'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/lib/colorls/flags.rb:31:in `process'
        from /home/patrick/.local/share/gem/ruby/3.0.0/gems/colorls-1.4.4/exe/colorls:26:in `<top (required)>'
        from /usr/bin/colorls:23:in `load'
        from /usr/bin/colorls:23:in `<main>'
```

## Steps to reproduce the error

- create a new folder and switch to it: `mkdir foo && cd foo`
- create a new git repo: `git init`
- install slidev: `npm init slidev@0.10.2`
- accept all defaults
- switch to folder slidev: `cd slidev`
- `lc` (`alias lc='colorls -lAG --sd --gs'`)

Let me know if you need more infos.

**Comment by jakbin** [comment]
same here


        Traceback (most recent call last):
	25: from /usr/local/bin/colorls:23:in `<main>'
	24: from /usr/local/bin/colorls:23:in `load'
	23: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/exe/colorls:26:in `<top (required)>'
	22: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/flags.rb:31:in `process'
	21: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/flags.rb:83:in `process_args'
	20: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/flags.rb:83:in `each'
	19: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/flags.rb:86:in `block in process_args'
	18: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:59:in `ls_dir'
	17: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:91:in `ls'
	16: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each_line'
	15: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `each'
	14: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/layout.rb:14:in `block in each_line'
	13: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:92:in `block in ls'
	12: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `ls_line'
	11: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `each_with_index'
	10: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:322:in `each'
	 9: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:323:in `block in ls_line'
	 8: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:316:in `fetch_string'
	 7: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:247:in `git_info'
	 6: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/core.rb:124:in `block in init_git_status'
	 5: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/git.rb:17:in `status'
	 4: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/git.rb:59:in `git_subdir_status'
	 3: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/git.rb:59:in `popen'
	 2: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/git.rb:66:in `block in git_subdir_status'
	 1: from /var/lib/gems/2.7.0/gems/colorls-1.4.4/lib/colorls/git.rb:22:in `block in status'
        /usr/lib/ruby/2.7.0/set.rb:339:in `add': can't modify frozen Hash: {"??"=>true} (FrozenError)

**Comment by avdv** [comment]
If you can, please try the fix in PR #478

**Comment by fharper** [comment]
This PR fixed the issue for me
